.github/workflows/dns.yaml

```diff
@@ -0,0 +1,47 @@
+on:
+  pull_request:
+jobs:
+  octodns-sync:
+    name: Run `octodns-sync` with production.yaml
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        working-directory: ./dns
+    outputs:
+      plan: ${{ steps.generate-plan.outputs.plan }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.12'
+      - run: pip install -r requirements.txt
+      - uses: solvaholic/octodns-sync@main
+        id: generate-plan
+        with:
+          config_path: dns/production.yaml
+        env:
+          CLOUDFLARE_TOKEN: ${{ secrets.CLOUDFLARE_TOKEN }}
```

Comment:
This begs the question: How secure is it? I suppose once again as secure as our branch restrictions?

Reply:
In which sense?

It will not deploy any changes that remove over 30% of entries, and we obviously do not even run the plan on fork PRs.

We could try enforce it so DNS changes have to go through a PR process and have a plan generated, or tighten our branch restrictions.

Reply:
My worry is just bad actors being able to tinker with our DNS. I think our staff are trustworthy, but with the current setup, over 150 people are able to tinker with our DNS, from my understanding:

![image](https://github.com/python-discord/infra/assets/22679924/abefb3c1-c383-4adf-8220-e42d9f2210d4)

Should we restrict these permissions?

Reply:
I have just checked on this.

We enforce PRs for the main branch as well as restricting that only members of python-discord/devops can push to this branch, so it should be sufficiently safe.

Reply:
What prevents changing the actions workflow from the branch such that it will deploy the changes on the branch? The token is available, or is the non-master token read-only?

Reply:
Hmmm this is a good point, I can split it into read-only & write tokens for when I add the deployment flow.